iothub/service/src/Common/Extensions/CommonExtensions.cs

```diff
@@ -17,12 +17,12 @@
 
 namespace Microsoft.Azure.Devices.Common
 {
-    public delegate bool TryParse<TInput, TOutput>(TInput input, bool ignoreCase, out TOutput output);
+    internal delegate bool TryParse<TInput, TOutput>(TInput input, bool ignoreCase, out TOutput output);
```

Comment:
We were using this helper only in the IotHubConnectionStringBuilder class. It has functions like IsNullOrWhiteSpace, RemoveWhitespace etc... They definitely don't need any xml comments. However this seems like a very good candidate to make internal. There is an off chance a customer is dependent on this but looks very highly unlikely to me. I want to bring this up here so that I can get some insight from others as well who have more context.

Reply:
These being extension methods on types such as strings and dictionaries, there might be a slight chance that users could have taken a dependency on these, for some logic in their application code. It wouldn't hurt to leave it as-is.

Reply:
I feel like customers might not be using this and should be safe. Is is good for us to not provide such extensions. This class is truly internal. But I definitely don't want to break customers if it is not acceptable. @drwill-ms what are your thoughts? Can we use our discretion and make such changes or not?

Reply:
I don't see a compelling enough reason to possibly break customers. Why take the chance?

Reply:
In the very least, we'd mark it as deprecated and eventually make it internal.

Reply:
That is good. Let me mark it as deprecated. We should surely give the impression that we will help support any kind of custom string extensions.